app/src/main/java/com/lagradost/quicknovel/providers/NovelFullProvider.kt

```diff
@@ -0,0 +1,236 @@
+package com.lagradost.quicknovel.providers
+
+import com.lagradost.quicknovel.*
+import org.jsoup.Jsoup
+import java.lang.Exception
+import kotlin.collections.ArrayList
+
+class NovelFullProvider : MainAPI() {
+    override val name = "NovelFull"
+    override val mainUrl = "https://novelfull.com"
+    override val hasMainPage = true
+
+    override val iconId = R.drawable.icon_novelfull
+
+    override val iconBackgroundId = R.color.white
+
+    override val tags = listOf(
+        Pair("All", "All"),
+        Pair("Shounen", "Shounen"),
+        Pair("Harem", "Harem"),
+        Pair("Comedy", "Comedy"),
+        Pair("Martial Arts", "Martial Arts"),
+        Pair("School Life", "School Life"),
+        Pair("Mystery", "Mystery"),
+        Pair("Shoujo", "Shoujo"),
+        Pair("Romance", "Romance"),
+        Pair("Sci-fi", "Sci-fi"),
+        Pair("Gender Bender", "Gender Bender"),
+        Pair("Mature", "Mature"),
+        Pair("Fantasy", "Fantasy"),
+        Pair("Horror", "Horror"),
+        Pair("Drama", "Drama"),
+        Pair("Tragedy", "Tragedy"),
+        Pair("Supernatural", "Supernatural"),
+        Pair("Ecchi", "Ecchi"),
+        Pair("Xuanhuan", "Xuanhuan"),
+        Pair("Adventure", "Adventure"),
+        Pair("Action", "Action"),
+        Pair("Psychological", "Psychological"),
+        Pair("Xianxia", "Xianxia"),
+        Pair("Wuxia", "Wuxia"),
+        Pair("Historical", "Historical"),
+        Pair("Slice of Life", "Slice of Life"),
+        Pair("Seinen", "Seinen"),
+        Pair("Lolicon", "Lolicon"),
+        Pair("Adult", "Adult"),
+        Pair("Josei", "Josei"),
+        Pair("Sports", "Sports"),
+        Pair("Smut", "Smut"),
+        Pair("Mecha", "Mecha"),
+        Pair("Yaoi", "Yaoi"),
+        Pair("Shounen Ai", "Shounen Ai"),
+        Pair("History", "History"),
+        Pair("Reincarnation", "Reincarnation"),
+        Pair("Martial", "Martial"),
+        Pair("Game", "Game"),
+        Pair("Eastern", "Eastern"),
+        Pair("FantasyHarem", "FantasyHarem"),
+        Pair("Yuri", "Yuri"),
+        Pair("Magical Realism", "Magical Realism"),
+        Pair("Isekai", "Isekai"),
+        Pair("Supernatural Source:Explore", "Supernatural Source:Explore"),
+        Pair("Video Games", "Video Games"),
+        Pair("Contemporary Romance", "Contemporary Romance"),
+        Pair("invayne", "invayne"),
+        Pair("LitRPG", "LitRPG"),
+        Pair("LGBT", "LGBT"),
+        Pair(
+            "Comedy Drama Romance Shounen Ai Supernatural",
+            "Comedy Drama Romance Shounen Ai Supernatural"
+        ),
+        Pair("Shoujo Ai", "Shoujo Ai"),
+        Pair("Supernatura", "Supernatura"),
+        Pair("Canopy", "Canopy")
+    )
+
+    override fun loadMainPage(
+        page: Int,
+        mainCategory: String?,
+        orderBy: String?,
+        tag: String?
+    ): HeadMainPageResponse {
+        val url = "$mainUrl/$tag/page/$page/"
```

Comment:
url gets defined later so this does nothing, please check the site how they do this

Reply:
@LagradOst I did it the same way I did with AllNovel